src/command/mcl.zig

```diff
@@ -1896,6 +2011,227 @@ fn mclWaitRecoverSlider(params: [][]const u8) !void {
     }
 }
 
+/// Add logging configuration for registers logging in the specified line
+fn addLogRegisters(params: [][]const u8) !void {
+    const line_name = params[0];
+    const line_idx = try matchLine(line_names, line_name);
+    const line = mcl.lines[line_idx];
+
+    var log: LogLine = undefined;
+
+    // Validate "axes" parameter
+    var axis_input_iterator = std.mem.tokenizeSequence(u8, params[1], ",");
+    while (axis_input_iterator.next()) |token| {
+        const axis_index = try std.fmt.parseInt(mcl.Station.Id, token, 0) - 1;
+        if (axis_index < 0 or axis_index > line.axes.len - 1) {
+            return error.InvalidAxis;
+        }
+        const station_index: Station.Index = @intCast(axis_index / 3);
+        log.stations[station_index] = true;
+    }
+
+    // Validate "registers" parameter
+    var reg_input_iterator = std.mem.tokenizeSequence(u8, params[2], ",");
+    outer: while (reg_input_iterator.next()) |token| {
+        inline for (@typeInfo(LogLine.RegisterType).@"enum".fields) |field| {
+            if (std.ascii.eqlIgnoreCase(field.name, token)) {
+                log.registers.set(@enumFromInt(field.value), true);
+                continue :outer;
+            }
+        }
+        return error.InvalidRegister;
+    }
+
+    var info_buffer: [64]u8 = undefined;
+    const prefix = "Ready to log registers: ";
+    @memcpy(info_buffer[0..prefix.len], prefix);
+    var buf_len = prefix.len;
+
+    var register_iterator = log.registers.iterator();
+    while (register_iterator.next()) |reg_entry| {
+        if (!log.registers.get(reg_entry.key)) continue;
+        @memcpy(info_buffer[buf_len .. buf_len + @tagName(reg_entry.key).len], @tagName(reg_entry.key));
+        @memcpy(info_buffer[buf_len + @tagName(reg_entry.key).len .. buf_len + @tagName(reg_entry.key).len + 1], ",");
+        buf_len += @tagName(reg_entry.key).len + 1;
+    }
+    std.log.info("{s}", .{info_buffer[0 .. buf_len - 1]});
+    log.status = true;
+    log_lines[line_idx] = log;
+}
+
+fn removeLogRegisters(params: [][]const u8) !void {
+    const line_name = params[0];
+    const line_idx = try matchLine(line_names, line_name);
+
+    if (log_lines[line_idx].status == false) {
+        std.log.err("Line is not configured for logging yet", .{});
+        return;
+    }
+
+    log_lines[line_idx].status = false;
+}
+
+fn resetLogRegisters(_: [][]const u8) !void {
+    for (0..line_names.len) |i| {
+        log_lines[i].status = false;
+    }
+}
+
+fn statusLogRegisters(_: [][]const u8) !void {
+    const stdout = std.io.getStdOut().writer();
```

Comment:
Minor: I'd actually suggest formatting everything into one string, and then emitting to log info.